TimeAlarms.cpp

```diff
@@ -242,7 +242,7 @@ void TimeAlarmsClass::serviceAlarms()
           Alarm[servicedAlarmId].updateNextTrigger();
         }
         if (TickHandler != NULL) {
-          (*TickHandler)();     // call the handler
+          TickHandler();     // call the handler
```

Comment:
This pull request was not merged because of this change.

Reply:
Hi, even though I'm not the author of the pull request, I'm still wondering why line 245 is preventing the PR from being merged. Would you mind elaborating? To me the proposed extension seem to be implemented exactly as it is in other libraries (e.g. the ubiquitous "PubSubClient" MQTT client). The syntax change of line 245 simply ensures that the calling syntax is compatible with both cases: plain old function pointer and std::function.

I'd quite like to see this extension implemented in TimeAlarms. It allows for more elegant OOP design.

Reply:
If you like you can have a try of my fork of the TimeAlarm. I rewrote it using the latest ESP 2.4 that is using Newlib C time functions.
https://github.com/jhagberg/ESP8266TimeAlarms

Reply:
Thanks for pointing that out! Personally I try as much as possible to use the libraries available through platform.io's package manager and treat those as the "canonical" versions. That's why I'm interested in contributing some minor ESP8266-relatex fixes. A second consideration is to use platform-agnostic libraries (AVR vs ESP) whenever possible.
But then again, having a rebased library specifically for the ESP8266's C library is quite nice, too. Thanks!

Reply:
Yes this code is not reviewed by anyone yet. But it will handle Timezones and Daylight savings very nicely without any extra library. And works for me for the moment at least!